namespaces/decentraland.md

```diff
@@ -0,0 +1,58 @@
+---
+namespace: urn:decentraland
+github-org: https://github.com/decentraland
+contact-email: developers@decentraland.org
+---
+
+# Description
+
+Decentraland is a virtual reality platform powered by the Ethereum blockchain. 
+
+# Resolver structure
+
+The structure of the resolver is defined by a hierarchy of content address resolvers.
+
+```yml
+urn:decentraland:{protocol}:(...)
+```
+
+The hierarchy is defined by the `{protocol}` component of the URN.
+
+## The Ethereum resolvers
+
+Decentraland works in Ethereum mainnet, ropsten and Matic. There are 4 registered roots for blockchain assets:
+
+- Ethereum Mainnet: `urn:decentraland:ethereum:{contract-or-alias}:{tokenId}`
```

Comment:
I'm not sure about `contract-or-alias`. `ens` could be used too. It is a common practice for resolving balances of addresses while using the web3 libs as ethers or web3: `getBalanceOf('nacho.eth')`.

If this is going to be only for assets, I will leave some suggestions but be my guess and choose whatever you feel comfortable with:

- `asset`
- `collection`
- `resource`

Reply:
For collections, we will have `itemId` in the pointer instead of `tokenId` which is a reference of a specific minted asset. I would suggest just using `id` for this.

Reply:
Aren't collections ERC721?

Reply:
are ERC721 compliant but what users upload to the content server for our collection are items. Tokens are items kind

Reply:
We can have different routes for different handlers, it wouldn't be a problem to have a handler for
- collections v2: `urn:decentraland:ethereum:collection:v1:(?<collection_address>0x[a-fA-F0-9]+):(?<tokenId>0x[a-fA-F0-9]+|\d+)`
- collections v1: `urn:decentraland:ethereum:collection:v1:(?<collection_address>0x[a-fA-F0-9]+):(?<id>[^:]+)`

We can try to find a generalization that scales in the future by being _very_ specific about the use cases, adding namespaces won't hurt. Or we can define a very loose schema `urn:decentraland:{protocol}:{registry}:{id}` and then deal with registry the way we can. If it is always hex numbers, it will be very difficult to differentiate one element from another.

Reply:
I can't see any issue by using `urn:{namespace}:{protocol}:{registry-or-contract-or-collection-or-alias}:{id}`. Have you found any?

P.S: I've used `registry-or-contract-or-collection-or-alias` because I don't want to enforce any. I will be ok with your decision.